docs/source/datamodules.rst

```diff
@@ -11,33 +11,63 @@ Data preparation in PyTorch follows 5 steps:
 A DataModule is simply a collection of a train_dataloader, val_dataloader(s), test_dataloader(s) along with the
 matching transforms and data processing/downloads steps required.
 
+.. code-block:: python
```

Comment:
why not test code?

Reply:
there are a ton of unnecessary doctests.

Reply:
What you mean by unnecessary, that there is no need for examples? Otherwise all examples shall be tested that they are aligned with actual code base...
cc: @awaelchli

Reply:
agree, this code block would be perfect for a doctest, and it is as simple as adding `.. testcode`. Even if we don't make any assertions here, Python will parse the code and run the import statements. It would help us keep the docs up-to-date with the api.

Reply:
yeah we do not need anything extra, but it checks syntax and eventually nb of passed arguments or kwargs